madara/crates/client/block_import/src/types.rs

```diff
@@ -218,3 +218,17 @@ pub struct BlockImportResult {
 
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct PendingBlockImportResult {}
+
+/// Output of a [`crate::reorg`] operation. Fields are designed after `ReorgData` from the v0.8 RPC
+/// spec (https://github.com/starkware-libs/starknet-specs/blob/6e5a577802a5e117da747296bed0545ee41f451d/api/starknet_ws_api.json#L348)
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ReorgData {
+    /// Hash of the first known block of the orphaned chain
+    pub starting_block_hash: Felt,
```

Comment:
why do we need block hashes here?
i think if you're just modifying the backend, you should probably just have a revert_to(block_n: u64) function in the backend, it would be the responsability of sync to catch a reorg.

Well, unless you're making a broadcast channel in the backend to listen to reorgs - but i thought you were still working on the database reverting part.

Reply:
The link above explains why -- it's to prepare for the reorg subscription RPC.

But it also gives more information than just block number, I'd rather provide as much context as possible to the caller.

It's also used in tests IIRC.